### PR TITLE
KIALI-936 Metrics queries can receive quantile parameters

### PR DESCRIPTION
**Pull request by jotak**
JIRA: https://issues.jboss.org/browse/KIALI-936

Related frontend PR: https://github.com/kiali/kiali-ui/pull/701

** Backwards incompatible? **
yes: Changes in REST API, https://github.com/kiali/kiali-ui/pull/701 is necessary for no regression

### CODE REVIEW

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-core-pr-e2e-test [#83](https://paste.fedoraproject.org/paste/03nILhsMfrj9yvCWU7ufOQ)
  * :red_circle: Istio-and-Kiali-on-Openshift [#[1373]](https://paste.fedoraproject.org/paste/N4n8ZjCr2hb6SH5wKe2S3A)
  * :heavy_check_mark: Istio-Upstream [#[249]](https://paste.fedoraproject.org/paste/BAqrgtmaVbSb31AHtkxFvg)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1291, 1292]](https://paste.fedoraproject.org/paste/cNnAskICtcfnkQ4QZoQ~aQ)
  * :heavy_check_mark: install-bookinfo [#[732]](https://paste.fedoraproject.org/paste/6sQTnaKLfBt1RD5Oaw9~iA)
  * :heavy_check_mark: Kiali-on-Openshift [#[743]](https://paste.fedoraproject.org/paste/ocjvyMvvRySkyLHsTjA6Ow)
  * :red_circle: run-kiali-e2e-tests [#[190]](https://paste.fedoraproject.org/paste/bUkDxrxumG4Va2M40Y8Fiw)


KIALI_NAME=kialicorepr514-istio-system

File: prometheus/metrics.go [hunk 1]
```diff
@@ -41,6 +43,8 @@ func (q *MetricsQuery) FillDefaults() {
 	q.Step = 15 * time.Second
 	q.RateInterval = "1m"
 	q.RateFunc = "rate"
+	q.Avg = true
+	q.Quantiles = []string{"0.5", "0.95", "0.99", "0.999"}
```
Comment (jshaughn) [inline prometheus/metrics.go]: This doesn't match the documented default.  Are you sure you want "0.999"?

Comment (jotak) [reply prometheus/metrics.go]: Oops, you're right (and doc is right, i'll remove 0.999 here)

Comment (jotak) [reply prometheus/metrics.go]: fixed. BTW I'm adding DNM label to sync with kiali/kiali-ui#701 review

File: handlers/metrics.go [hunk 2]
```diff
@@ -52,6 +52,17 @@ func extractMetricsQueryParams(r *http.Request, q *prometheus.MetricsQuery) erro
 	if filters, ok := queryParams["filters[]"]; ok && len(filters) > 0 {
 		q.Filters = filters
 	}
+	if quantiles, ok := queryParams["quantiles[]"]; ok && len(quantiles) > 0 {
+		q.Quantiles = quantiles
```
Comment (lucasponce) [inline handlers/metrics.go]: Minor one, if I put a wrong quantile here, a bad request (400) is sent ? or it falls into the default quantile value ?
(perhaps this is coded later, but I leave question at the handler)

Comment (jotak) [reply handlers/metrics.go]: There's actually no validation at all neither here nor elsewhere, I'm going to add it.

**Comment by jotak** [comment]
> I think this needs to be merged together with the UI at same time, to not break the changes on the metric types.

@lucasponce yes, I'll merge them together when both are reviewed

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-core-pr-e2e-test [#96](https://paste.fedoraproject.org/paste/ov52oHrEh2susZR5dkkYOg)
  * :red_circle: Istio-and-Kiali-on-Openshift [#[1407]](https://paste.fedoraproject.org/paste/3A9A10sKj7UXksvGjXhHSQ)
  * :red_circle: Istio-Upstream [#[280]](https://paste.fedoraproject.org/paste/h0OToqCdKFzZrYndFi26yg)


KIALI_NAME=kialicorepr514-istio-system

**Comment by jotak** [comment]
As stated by @israel-hdez I will change the default values for quantiles to be empty: https://github.com/kiali/kiali-ui/pull/701#pullrequestreview-159099196

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-core-pr-e2e-test [#103](https://paste.fedoraproject.org/paste/2kPTuvs~gUBsNxPx~nRZ8w)
  * :red_circle: Istio-and-Kiali-on-Openshift [#[1424]](https://paste.fedoraproject.org/paste/0ofMqXuys~r1T-rN~8eWCQ)
  * :heavy_check_mark: Istio-Upstream [#[296]](https://paste.fedoraproject.org/paste/N1DCsCUi6sQC91UZI9m3bg)
  * :heavy_check_mark: Kiali-Istio-Checker [#[1392, 1393]](https://paste.fedoraproject.org/paste/ciWXwUJQDf4DyMoPmrfbow)
  * :heavy_check_mark: install-bookinfo [#[787]](https://paste.fedoraproject.org/paste/0lcBWOYXtgmLAeiovZ8P~g)
  * :heavy_check_mark: Kiali-on-Openshift [#[789]](https://paste.fedoraproject.org/paste/JSE1SZg~0c~UItEeNqbzCQ)
  * :red_circle: run-kiali-e2e-tests [#[232]](https://paste.fedoraproject.org/paste/4G8HV1DSto1Pdt5IqWRnAQ)


KIALI_NAME=kialicorepr514-istio-system

**Pull request merged**